### PR TITLE
Make 'No' button default in Unknown Certificate dialog

### PR DESCRIPTION
**Pull request by erikjv**
Fixes: #11531

### CODE REVIEW

**Comment by TheOneRing** [comment]
The ui tests have multiple crashes in the tls dialog, please have a look.

**Review by TheOneRing — CHANGES_REQUESTED** [review]
^

File: src/gui/tlserrordialog.cpp [hunk 1]
```diff
@@ -40,6 +40,12 @@ TlsErrorDialog::TlsErrorDialog(const QList<QSslError> &sslErrors, const QString
 
     connect(_ui->buttonBox, &QDialogButtonBox::accepted, this, &TlsErrorDialog::accept);
     connect(_ui->buttonBox, &QDialogButtonBox::rejected, this, &TlsErrorDialog::reject);
+
+    // Try to set the No-button (the reject button) as the default
+    if (QPushButton *noButton = _ui->buttonBox->button(QDialogButtonBox::No)) {
+        Q_ASSERT(_ui->buttonBox->buttonRole(noButton) == QDialogButtonBox::RejectRole || _ui->buttonBox->buttonRole(noButton) == QDialogButtonBox::NoRole);
```
Comment (TheOneRing) [inline src/gui/tlserrordialog.cpp:46]: Hmm according to https://doc.qt.io/qt-6/qdialogbuttonbox.html#ButtonRole-enum it is supposed to always be `NoRole`?

Comment (erikjv) [reply src/gui/tlserrordialog.cpp:46]: But how can clicking the "No" button then reject the dialog?

Comment (erikjv) [reply src/gui/tlserrordialog.cpp]: I removed the assert, it's superfluous to check the `NoRole` on the "No" button.

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**